feature_apps/src/main/java/cm/aptoide/pt/feature_apps/data/AptoideBundlesRepository.kt

```diff
@@ -0,0 +1,67 @@
+package cm.aptoide.pt.feature_apps.data
+
+import cm.aptoide.pt.feature_apps.domain.*
+import kotlinx.coroutines.flow.*
+import timber.log.Timber
+
+internal class AptoideBundlesRepository(
+  private val widgetsRepository: WidgetsRepository,
+  private val appsRepository: AppsRepository,
+) :
+  BundlesRepository {
+  override fun getHomeBundles(): Flow<BundlesResult> = flow {
+    val bundlesFlow = widgetsRepository.getStoreWidgets()
+      .flatMapConcat {
+        if (it is Result.Success) {
+          return@flatMapConcat it.data.asFlow()
+        } else {
+          return@flatMapConcat flow { BundlesResult.Error(IllegalStateException()) }
+        }
+      }
+      // https://aptoide.atlassian.net/browse/APP-954
+      // this should be a merge (to parallelize calls) the issue here atm is preserving order.
+      // Which with merge only it isn't achieved.
+      // Because we are only polling 5 bundles it isn't a big deal... but should be optimized.
+      // This might be a good starting point:
+      // https://discuss.kotlinlang.org/t/how-to-merge-flows-while-preserving-the-order/22002/2
+      .flatMapConcat { widget ->
+        when (widget.type) {
+          WidgetType.APPS_GROUP -> appsRepository.getAppsList(widget.view.toString()).map {
+            return@map mapAppsWidgetToBundle(it, widget)
+          }.catch { }
+          else -> appsRepository.getAppsList(widget.title).map {
```

Comment:
why do you use the title here ? isnt the url always on the view attribute?

Reply:
this was just to force error. I had to use a default branch for when widget type is not apps_group because I still don't have the implementation of some widgets (promotional, earn 2 play and so on). This will disappear when all widgets are implemented. I have removed the title and I am just passing the empty string. again this will be removed.